src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private int threshold;
+    private Node<K, V> [] table;
+
+    class Node<K, V> {
+        private final int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key) % table.length;
```

Comment:
You have a potential bug here. The hash code can be negative, and using the modulus operation `%` with a negative number can also result in a negative number. This will cause an `ArrayIndexOutOfBoundsException`. To fix this, you can use `Math.abs(hash(key)) % table.length`.